ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -429,26 +431,34 @@ ledgerDbPush cfg ap db =
 
 -- | Push a bunch of blocks (oldest first)
 ledgerDbPushMany :: (ApplyBlock l blk, Monad m, c)
-                 => LedgerDbCfg l
+                 => (UpdateLedgerDbTraceEvent blk -> m ())
+                 -> LedgerDbCfg l
                  -> [Ap m l blk c] -> LedgerDB l -> m (LedgerDB l)
-ledgerDbPushMany = repeatedlyM . ledgerDbPush
+ledgerDbPushMany trace = repeatedlyM . pushAndTrace
+  where
+    pushAndTrace cfg ap db = do
+      trace StartedPushingBlockToTheLedgerDb
```

Comment:
Do we trace somewhere else some details about the block we're validating? If not we might want to add these details here. 

Also, I don't know if we should state we're pushing a block to the ledger DB, since that depends upon the block being valid.

Reply:
> Do we trace somewhere else some details about the block we're validating? If not we might want to add these details here.

What we have here is just `ap`, to get more details we would have to call `resolveBlock r` for some of the `ap`s. This means we would have to do it twice (once for tracing and once in `applyBlock`). 
Alternative would be to push the tracing into `Ap` (we've went that path in this PR as you recall). Or pushing the `trace` function down to `applyBlock`. But that means that `trace` would change the signature of `ledgerDbPush`, a function that is used in other places

Reply:
> Also, I don't know if we should state we're pushing a block to the ledger DB, since that depends upon the block being valid.

Function is called `ledgerDBPush` (not `tryLedgerDBPush`) which got me confused, but I see that we throw an error for some data constructors of `Ap`. We can:
a) rename it, though I can not think of a name that will be a right one
b) push the trace to `applyBlock` and fire events accordingly

Reply:
Decisons during PP:
1. we will add `RealPoint blk` to the `StartedPushingBlockToTheLedgerDb` event
2. we will remove the `PushedBlockToTheLedgerDb` event
3. we will add progress to the `UpdateLedgerDbTraceEvent` so that we can track progress in cardano-node